packages/@webex/plugin-meetings/src/common/taskQueue.ts

```diff
@@ -0,0 +1,86 @@
+import {Defer} from '@webex/common';
+import window from 'global/window';
+
+type Task<TResult> = () => TResult;
+type TaskItem<TResult> = {task: Task<TResult>; defer: Defer};
+type CancelScheduler = () => void;
+type TaskQueueScheduler = (fn: () => void) => CancelScheduler;
+type TaskQueueOptions = {
+  scheduler?: TaskQueueScheduler;
+};
+
+const timout = (task) => setTimeout(task, 1000 / 60);
+const defaultSchedulerFactory = (): TaskQueueScheduler => {
+  const scheduler = window.requestIdleCallback || window.requestAnimationFrame || timout;
+  const cancel = window.cancelIdleCallback || window.cancelAnimationFrame || clearTimeout;
+
+  return (task) => {
+    const handler: any = scheduler(task);
+
+    return () => cancel(handler);
+  };
+};
+
+/**
+ * Execute tasks asynchronously in fifo order.
+ *
+ * Task scheduler can be customised.
+ * The default scheduler is the `requestIdleCallback` API which falls back to `requestAnimationFrame` or `setTimeout`
+ *
+ * Ideal for breaking up long running tasks.
+ * @link https://web.dev/long-tasks-devtools/
+ * @returns {TaskQueue}
+ */
+export default class TaskQueue<TResult = void> {
+  private readonly scheduler;
+  private readonly queue: TaskItem<TResult>[] = [];
+  private cancelSchedule?: CancelScheduler;
+  private running = false;
+  private currentTask?: TaskItem<TResult>;
+
+  /**
+   * @constructs TaskQueue
+   */
+  constructor({scheduler}: TaskQueueOptions = {}) {
+    this.scheduler = scheduler || defaultSchedulerFactory();
+  }
+
+  private run = () => {
+    if (!this.running && this.queue.length > 0) {
+      this.running = true;
+      this.currentTask = this.queue.shift();
+      this.cancelSchedule = this.scheduler(() => {
+        const result = this.currentTask.task();
```

Comment:
if the task throws the promise will never be resolved. Shouldn't we be rejecting the defer in the event the task throws?